src/pie_utils/processor/document/split_into_partitions.py

```diff
@@ -0,0 +1,123 @@
+                span = Span(start=previous_start, end=end)
+                yield span
+
+            previous_start = match.start()
+
+    if previous_start is not None:
+        end = len(document.text)
+        span = Span(start=previous_start, end=end)
+        yield span
+
+
+class SplitDocumentToPartitions(WithStatistics):
+    def __init__(
+        self,
+        pattern: str,
+        label_group_id: int | None = None,  # = 1,
+        label_whitelist: list[str] | None = None,
+        skip_initial_partition: bool = False,  # = True
+        collect_statistics: bool = False,
+    ):
+        self.label_group_id = label_group_id
+        self.label_whitelist = label_whitelist
+        self.skip_initial_partition = skip_initial_partition
+        self.matcher = re.compile(pattern).finditer
+        self.collect_statistics = collect_statistics
+        self.reset_statistics()
+
+    def reset_statistics(self):
+        self._statistics: dict[str, Any] = {
+            "partition_lengths": [],
+            "num_partitions": [],
+            "partition_texts": [],
+        }
+
+    def show_statistics(self, description: str | None = None):
+        description = description or "Statistics"
+        logger.info(f"{description}: \n{json.dumps(self._statistics, indent=2)}")
+
+    def update_statistics(self, key: str, value: int | str | list):
+        if self.collect_statistics:
+            if isinstance(value, list):
+                self._statistics[key] += value
+            elif isinstance(value, str) or isinstance(value, int):
+                self._statistics[key].append(value)
+            else:
+                raise TypeError(
+                    f"type of given key [{type(key)}] or value [{type(value)}] is incorrect."
+                )
+
+    def __call__(self, document: DocumentWithPartition):
+        partitions_for_doc: list[Span] = []
+        partition_lengths = []
+        partition_texts = []
+        for partition in get_partitions_with_matcher(
+            document,
+            matcher=self.matcher,
+            skip_initial_partition=self.skip_initial_partition,
+            label_whitelist=self.label_whitelist,
+            label_group_id=self.label_group_id,
+        ):
+            # just a sanity check
+            for s in partitions_for_doc:
+                if have_overlap((s.start, s.end), (partition.start, partition.end)):
+                    logger.error(f"overlap: {partition} with {s}")
+
+            document.partition.append(partition)
+            partitions_for_doc.append(partition)
+            partition_texts.append(document.text[partition.start : partition.end])
+            partition_lengths.append(partition.end - partition.start)
+
+        if self.collect_statistics:
+            self.update_statistics("num_partitions", len(document.partition))
+            self.update_statistics("partition_lengths", partition_lengths)
+            self.update_statistics("partition_texts", partition_texts)
```

Comment:
Ah ok, that was not clear in the issue description. I meant "This should also collect the distribution of the lengths of [...] the full texts (to compare against)" ;)

Reply:
by full texts, do you mean length of full document text ?

Reply:
yes, to see the effect of the partitioning when comparing with `partition_lengths`

Reply:
fixed in [eeb2b72](https://github.com/ArneBinder/pie-utils/pull/21/commits/eeb2b72429079892d22a699f11d496f42ec3922e)

Reply:
I think there is an error in the implementation. can you add a check to the test that checks the actual content to `document_lengths` (not just the sum)?

Reply:
added this check in [ed39659](https://github.com/ArneBinder/pie-utils/pull/21/commits/ed39659f1b3ba9e0ef2235a6c25474a558f999d9). Is this what you mean by implementation error or something else ?

Reply:
ok, sorry, I misread the code, my bad. But still the check looks better now :)